inji-web/src/__tests__/pages/HomePage.test.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import { screen, waitFor, fireEvent } from '@testing-library/react';
+import { HomePage } from '../../pages/HomePage';
+import { reduxStore } from '../../redux/reduxStore';
+import { storeIssuers } from '../../redux/reducers/issuersReducer';
+import { renderWithRouter, mockusei18n,mockUseFetch,mockUseToast } from '../../test-utils/mockUtils';
+
+mockusei18n();
+const mockUseFetchHook = jest.fn();
+mockUseFetch();
+mockUseToast();
+
+//todo : extract the local method to mockUtils, which is added to bypass the routing problems
+jest.mock('react-toastify', () => ({
+  toast: {
+    error: jest.fn(),
+  },
+}));
+
+describe('Testing the Layout of HomePage', () => {
+  test('check if the layout is matching with the snapshots', () => {
+    mockUseFetchHook.mockReturnValue({ state: 'DONE', fetchRequest: jest.fn() });
+    const { asFragment } = renderWithRouter(<HomePage />);
+    expect(asFragment()).toMatchSnapshot();
+  });
+});
+
+describe('Testing the Functionality of HomePage', () => {
+  beforeEach(() => {
+    mockusei18n();
+    reduxStore.dispatch(storeIssuers([
+      {
```

Comment:
extract this to mockObject.tsx

Reply:
Extracting the Issuer Object from mockObjects.tsx